client/fingerprint_manager.go

```diff
@@ -361,6 +348,21 @@ func (fm *FingerprintManager) fingerprintDriver(name string, f fingerprint.Finge
 		fm.nodeLock.Unlock()
 	}
 
+	if !response.Detected {
```

Comment:
I think you want to also guard on `!DriverInfo.Healthy` otherwise the periodic fingerprinter will constantly override the health status.

Reply:
Where does the `!DriverInfo.Healthy` come from in this case? Not sure I completely follow, but if we can prevent overriding in the cases we want without causing flapping, that would be ideal.

Reply:
Essentially I am saying if the current driver info for this driver is unhealthy we don't need to set it to unhealthy and override the health description